### DISCUSSION TITLE
running SPAdes on paired end reads of different length didn't generate scaffolds.fasta

### DISCUSSION THREAD

**Discussion by AlumaEtzion** [ablab/spades#1232] [Q&A]
### Description of bug

I used spades to assemble paired end reads that aren't of the same length, my R1 file contains 50bp reads and my R2 file contains 26bp reads. The reads are in the same order in both files. 
I got a contigs.fasta file in my results but didn't get scaffolds.fasta, and I wanted to ask why? In addition, does the lack of scaffolds.fasta file indicate a bigger problem and my results (the contigs.fasta file) aren't reliable? 

I got this errors in my log file, but I didn't understand if they are related to my problem. or how I can fix them:
=== Error correction and assembling warnings:
 * 0:00:57.431    58M / 581M  WARN    General                 (pair_info_count.cpp       : 358)   Unable to estimate insert size for paired library #0
 * 0:00:57.431    58M / 581M  WARN    General                 (pair_info_count.cpp       : 364)   None of paired reads aligned properly. Please, check orientation of your read pairs.
 * 0:00:57.431    58M / 581M  WARN    General                 (repeat_resolving.cpp      :  81)   Insert size was not estimated for any of the paired libraries, repeat resolution module will not run.
======= Warnings saved to /sci/backup/morani/lab/Projects/Aluma/Hadza_genomic/subsampled_results/spades_results/paired/ERR9631799_sub_samp_2.5M_50_26bp_spades_paired_output/warnings.log

Thank you

### spades.log

[spades.log](https://github.com/ablab/spades/files/13828833/spades.log)

### params.txt

[params.txt](https://github.com/ablab/spades/files/13828835/params.txt)

### SPAdes version

3.15.5

### Operating System

Linux-5.10.104-aufs-3-x86_64-with-debian-10.10

### Method of SPAdes installation

binaries

### No errors reported in spades.log

- [X] Yes

**Comment by asl** [comment]
Hello

You left reads are short and therefore the largest k-mer length used is only k=33. However, your right reads is terrible short – shorter than k-mer length. As a result, they could not be used in repeat resolution process hence the warnings you've received.

The only way to "fix" the problem is to provide a decent input data.

**Reply by AlumaEtzion** [reply]
Thank you for your response.

When I ran spades on the same sample but with both reads of same length of 25bp it was able to produce a scaffolds.fasta file and I didn't get those warnings (I did get other warnings: === Error correction and assembling warnings:
 * 0:00:24.130    79M / 188M  WARN    General                 (kmer_coverage_model.cpp   : 218)   Too many erroneous kmers, the estimates might be unreliable
 * 0:00:24.747    79M / 188M  WARN    General                 (kmer_coverage_model.cpp   : 366)   Failed to determine erroneous kmer threshold. Threshold set to: 64
 * 0:00:50.427    73M / 459M  WARN    General                 (simplification.cpp        : 504)   The determined erroneous connection coverage threshold may be determined improperly ).
 
So I think there is more to it than only the shortness of my data.

My main concern isn't the lack of a scaffolds.fasta file in the run I described earlier, but whether or not I can trust the results that I did get in the contigs.fasta file. Does the lack of  scaffolds.fasta file indicate a bigger problem that might make my results unreliable?

Thank you again

**Reply by asl** [reply]
> So I think there is more to it than only the shortness of my data.

Not quite. As I already mentioned: you right reads could not be used in repeat resolution process with the k-mer length chosen hence the warnings you've received.

> Does the lack of scaffolds.fasta file indicate a bigger problem that might make my results unreliable?

Absolutely. I would not trust any results obtained from 25 bp reads. But still, you just get the output w/o repeat resolution being run. It's a normal assembler output
